bigtable/client/detail/bulk_mutator.cc

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/detail/bulk_mutator.h"
+
+#include <numeric>
+
+#include <bigtable/client/rpc_retry_policy.h>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace detail {
+
+namespace btproto = google::bigtable::v2;
+
+BulkMutator::BulkMutator(std::string const &table_name,
+                         IdempotentMutationPolicy &policy, BulkMutation &&mut) {
+  mutations.set_table_name(table_name);
```

Comment:
The implementation of this method could use some internal comments. I am not an expert on this, and I'm having trouble following what's what here and why; it would be nice to have some implementation guidance as to the rationale / high-level explanation of what's what here.

Reply:
Done.